### PR TITLE
docs(syslog source): Use generated docs for configuration

### LINKED ISSUE #16097: Convert the `syslog` source to use auto-generated docs

### PR DESCRIPTION
**Pull request by StephenWakely**
Closes #16097 
Epic #14998 

Signed-off-by: Stephen Wakely <fungus.humungus@gmail.com>

### CODE REVIEW

File: src/sources/syslog.rs [hunk 1]
```diff
@@ -71,7 +72,10 @@ pub struct SyslogConfig {
 pub enum Mode {
     /// Listen on TCP.
     Tcp {
-        /// The socket address to listen for connections on.
+        /// The socket address to listen for connections on, or `systemd{#N}` to use the Nth socket passed by
+        /// systemd socket activation.
+        ///
+        /// If a socket address is used, it _must_ include a port.
         address: SocketListenAddr,
```
Comment (neuronull) [inline src/sources/syslog.rs:79]:
```suggestion
        #[configurable(derived)]
        address: SocketListenAddr,
```

File: src/sources/syslog.rs [hunk 2]
```diff
@@ -83,6 +87,7 @@ pub enum Mode {
         /// The size, in bytes, of the receive buffer used for each connection.
```
Comment (neuronull) [inline src/sources/syslog.rs:87]:
```suggestion
        /// The size of the receive buffer used for each connection.
```

File: src/sources/syslog.rs [hunk 3]
```diff
@@ -97,6 +102,7 @@ pub enum Mode {
         /// The size, in bytes, of the receive buffer used for the listening socket.
```
Comment (neuronull) [inline src/sources/syslog.rs:102]:
```suggestion
        /// The size of the receive buffer used for the listening socket.
```

File: src/sources/syslog.rs (hunk 1 above)
Comment (StephenWakely) [reply src/sources/syslog.rs:79]: Oh, nice!

Comment (neuronull) [reply src/sources/syslog.rs:79]: 😃

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**